### ISSUE TITLE
Implement GraphHookTest interface on all GraphHook tests

### ISSUE THREAD

**Issue by m55624** [gchq/Gaffer#2073]
Currently there are only a few GraphHookTests that actually implement the correct interfaces.  All GraphHooks should implement this interface and correctly test everything, in addition to the extra tests.

For a correct example see AddOperationsToChainTest.java.
For an incorrect example see NamedViewResolverTest.java.

**Comment by n288TJYRX** [comment]
Can you assign this to me? I think I should be able to do this

**Comment by m55624** [comment]
We can't currently assign to you but feel free to pick it up.  When it's done just open the PR and I will have a read through.  If you're struggling at all let me know here and I can help.  You will need to sign the CLA when you open the PR. Thanks

**Comment by n288TJYRX** [comment]
Hi, I'm struggling with an error. I've made NamedViewResolverTest extend GraphHookTest but it fails to pass the shouldHaveJsonPropertyAnnotation test of JSONSerialisationTest.java. It says it is "Missing JsonPropertyOrder annotation on class"

**Comment by n288TJYRX** [comment]
For Log4jLogger I can't use fromJson to load the class from file, because the file is not json. The file is log4j.xml

**Comment by n288TJYRX** [comment]
There are no json files for the classes NamedViewResolver, NamedOperationResolver or UpdateViewHook should I just write "new ClassName()" in the getTestObject() methods?

**Comment by m55624** [comment]
Hi, so for the getTestObject method yeah just put a new instance of the class.

For example with the NamedOperationResolver GraphHook It will look something like this (obviously with the other current tests in there too):

```
public class NamedOperationResolverTest extends GraphHookTest<NamedOperationResolver> {
      public NamedOperationResolverTest() {
            super(NamedOperationResolver.class);
      }
      @Override
      protected NamedOperationResolver getTestObject() {
           return new NamedOperationResolver();
      }
}
```

For the shouldHaveJsonPropertyAnnotation test method to work this line will need to be added to the GraphHook class:
```@JsonPropertyOrder(alphabetic = true)```
You can go ahead and add these lines within this ticket.

Not sure what you mean about the Log4jlogger bits, I've just tested and it seems to be alright using the same implementation as above.

Thanks very much for this, let me know if there are any other issues.

**Comment by n288TJYRX** [comment]
Okay I've tried adding the annotation to the GraphHook interface, but the classes still don't pass the shouldHaveJsonPropertyAnnotation test. I think this is because GraphHook is an interface and not a class and you cannot inherit annotations from interfaces https://stackoverflow.com/questions/4745798/why-java-classes-do-not-inherit-annotations-from-implemented-interfaces
All I can think of is to add the annotation to each of these GraphHook tests that don't pass, or all of them. Or turn the GraphHook interface into a class. At the moment I've just added the annotation to all of the GraphHook tests.
